app/uploaders/logo_uploader.rb

```diff
@@ -0,0 +1,49 @@
+class LogoUploader < BaseUploader
+  MAX_FILE_SIZE = 3 # Megabytes
+  EXTENSION_ALLOWLIST = %w[svg png jpg jpeg jpe].freeze
+  IMAGE_TYPE_ALLOWLIST = %i[svg png jpg jpeg jpe].freeze
+  CONTENT_TYPE_ALLOWLIST = %w[image/svg+xml image/png image/jpg image/jpeg].freeze
+
+  def store_dir
+    "uploads/logos/"
+  end
+
+  def extension_allowlist
+    EXTENSION_ALLOWLIST
+  end
+
+  def image_type_whitelist
+    # this is needed by CarrierWave::BombShelter
+    IMAGE_TYPE_ALLOWLIST
+  end
+
+  def size_range
+    1..(MAX_FILE_SIZE.megabytes)
+  end
+
+  def content_type_allowlist
+    CONTENT_TYPE_ALLOWLIST
+  end
+
+  def filename
+    # random_string in the filename to avoid caching issues
+    "original_logo_#{random_string}.#{file.extension}" if original_filename
+  end
+
+  version :resized_logo, if: :not_svg? do
```

Comment:
Suggestion: `version` also has an `unless` option and I think `unless: :svg?` reads better than `if: not_svg?`.

Reply:
@citizen428 I did initially try unless: :svg? as follows: 

```
version :resized_logo, unless: :svg? do
```

```
  def svg?(file)
    put "code executes in here"
    file.content_type.include?("svg")
  end
 ```

However, for some reason it just was not calling the  `svg?` method, hence the `puts` was never getting written. [The Carrierwave source code](https://github.com/carrierwaveuploader/carrierwave/blob/7a40ef7c4d5f3033f0f8401323f80bde14ca72b9/lib/carrierwave/uploader/versions.rb#L179-L193) does accept `unless` with versions so honestly I'm not sure why it's not working unless there's something really obvious that I'm missing.

Reply:
It's really not that important, don't spend any more time on it than you already have.